### ISSUE TITLE
Can not load STK file 

### ISSUE THREAD

**Issue by htyeim** [yeesian/ArchGDAL.jl#170]
I downloaded `FLEVOL.STK` (https://earth.esa.int/documents/653194/658149/AIRSAR_Flevoland) file from https://earth.esa.int/web/polsarpro/data-sources/sample-datasets
Then used ArchGDAL to load it with:

```
using ArchGDAL
const AG = ArchGDAL
dataset = AG.read("FLEVOL.STK")
```

```
julia> dataset = AG.read("FLEVOL.STK")
GDAL Dataset (Driver: AirSAR/AirSAR Polarimetric Image)
File(s): 
  FLEVOL.STK

Dataset (width x height): 1024 x 750 (pixels)
Number of raster bands: 6
  Error showing value of type ArchGDAL.IDataset:
ERROR: KeyError: key GDAL.GDT_CFloat32 not found
Stacktrace:
 [1] getindex at ./dict.jl:467 [inlined]
 [2] ArchGDAL.IRasterBand(::Ptr{Nothing}; ownedby::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/types.jl:188
 [3] getband at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:606 [inlined]
 [4] show(::IOContext{REPL.Terminals.TTYTerminal}, ::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/base/display.jl:24
 [5] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::ArchGDAL.IDataset) at ./multimedia.jl:47
 [6] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:214
 [7] display(::REPL.REPLDisplay, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:218
 [8] display(::Any) at ./multimedia.jl:328
 [9] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [10] invokelatest at ./essentials.jl:709 [inlined]
 [11] print_response(::IO, ::Any, ::Bool, ::Bool, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:238
 [12] print_response(::REPL.AbstractREPL, ::Any, ::Bool, ::Bool) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:223
 [13] (::REPL.var"#do_respond#54"{Bool,Bool,REPL.var"#64#73"{REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.Prompt})(::Any, ::Any, ::Any) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:822
 [14] #invokelatest#1 at ./essentials.jl:710 [inlined]
 [15] invokelatest at ./essentials.jl:709 [inlined]
 [16] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/LineEdit.jl:2355
 [17] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.5/REPL/src/REPL.jl:1144
 [18] (::REPL.var"#38#42"{REPL.LineEditREPL,REPL.REPLBackendRef})() at ./task.jl:356
```

The error above seems related to display the dataset, so I add `;` then check the propertory of `dataset`
```
julia> using ArchGDAL

julia> const AG = ArchGDAL
ArchGDAL

julia> dataset = AG.read("FLEVOL.STK");

julia> AG.getdriver(dataset)
Driver: AirSAR/AirSAR Polarimetric Image

julia> AG.nraster(dataset)
6

julia> AG.width(dataset)
1024

julia> AG.height(dataset)
750

julia> gt = AG.getgeotransform(dataset)
ERROR: Failed to get geotransform
Stacktrace:
 [1] error(::String) at ./error.jl:33
 [2] macro expansion at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/utils.jl:22 [inlined]
 [3] getgeotransform! at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:642 [inlined]
 [4] getgeotransform(::ArchGDAL.IDataset) at /Users/t/.julia/packages/ArchGDAL/6e9sT/src/dataset.jl:646
 [5] top-level scope at REPL[12]:1

```
The package did get some information of the file, but failed to load the other.

The python package of GDAL can load the file correctly:
```
In [1]: from osgeo import gdal

In [2]: dataset = gdal.Open("FLEVOL.STK")

In [3]: dataset.RasterCount
Out[3]: 6

In [4]: dataset.GetRasterBand(1)
Out[4]: <osgeo.gdal.Band; proxy of <Swig Object of type 'GDALRasterBandShadow *' at 0x0000021E23AC8FC0> >

In [5]: a = dataset.GetRasterBand(1)

In [6]: r = a.ReadAsArray()
```

The GDAL itself support this type of file (https://gdal.org/drivers/raster/airsar.html).
Is it a problem related to ArchGDAL.jl?

**Comment by yeesian** [comment]
Yeah, it is, thanks for the report!

It looks like we need to update the following maps: https://github.com/yeesian/ArchGDAL.jl/blob/c1d9fafeec7b42fb63194bdf720f7e8865b304b7/src/types.jl#L257-L276 with something like 
```julia
GDAL.GDT_CFloat32    => Float32,
```
and
```julia
 Float32     => GDAL.GDT_CFloat32,
```
respectively.

**Comment by felixcremer** [comment]
We have two separate issues with loading the STK file. The first one is that the complex data type is not supported and that will be fixed by #208. 
The second one is, that currently AG.getgeotransform! throws an Error, when the underlying GDAL call returns not CE_None but it still puts the standard geotransform into the passed vector. 
One way to circumvent this problem is to use the AG.getgeotransform! function inside of a try catch block to still get the value in the passed vector, but to not get the error.

**Comment by yeesian** [comment]
> The second one is, that currently AG.getgeotransform! throws an Error, when the underlying GDAL call returns not CE_None but it still puts the standard geotransform into the passed vector.

Is this the same issue as #190?
